### DISCUSSION TITLE
sku name problem

### DISCUSSION THREAD

**Discussion by nsabidi** [opentofu/opentofu#2201] [Q&A] [answered]
Hi,
I have the error when I make tofu plan 
Planning failed. OpenTofu encountered an error while generating this plan.

╷
│ Error: invalid value for sku_name (This is not a valid Api Management sku name.)
│
│   with azurerm_api_management.main,
│   on maint.tf line 11, in resource "azurerm_api_management" "main":
│   11:   sku_name            = var.sku_name

in my variable.tf  I define 

variable "sku_name" {
description = "The SKU of the API Management instance."
type        = string
default     = "standard" 

I know that I can use "Consumption", "Developer", "Basic", "Standard" and "Premium" as default but none of them work and I have the same error.  I Use previously-installed hashicorp/azurerm v4.10.0 for tofu. 
Who can help me to handle this problem

Thank you in advance.

Dieudonné

**Comment by ollevche** [comment, marked as answer]
Hey @nsabidi! 

This part of validation is on the provider side so it is Azure-specific. I believe you also need to specify a capacity for `sku_name` field, e.g. `Developer_1`. Not sure if that's the provider you use, but here are [the docs](https://search.opentofu.org/provider/terraform-providers/azurerm/latest/docs/resources/api_management):

```
sku_name - (Required) sku_name is a string consisting of two parts separated by an underscore(_). The first part is the name, valid values include: Consumption, Developer, Basic, Standard and Premium. The second part is the capacity (e.g. the number of deployed units of the sku), which must be a positive integer (e.g. Developer_1).
```

Hope that helps!

**Reply by nsabidi** [reply]
thanks olllevche I tried to use the capacity. I still have the problem

**Reply by ollevche** [reply]
@nsabidi could you share the error message you get? It might be a different one
